platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -301,7 +301,12 @@ - (void)startWithQueue:(dispatch_queue_t)queue overlayHandler:(MGLMapSnapshotOve
     if (!completion) {
         return;
     }
-    if (!mbgl::Scheduler::GetCurrent()) {
+    
+    // Ensure that offline storage has been initialized on the main thread, as MGLMapView and MGLOfflineStorage do when used directly.
+    // https://github.com/mapbox/mapbox-gl-native-ios/issues/227
+    if ([NSThread.currentThread isMainThread]) {
+        (void)[MGLOfflineStorage sharedOfflineStorage];
+    } else {
         [NSException raise:NSInvalidArgumentException
                     format:@"startWithQueue:completionHandler: must be called from a thread with an active run loop."];
```

Comment:
Should we change this exception to something like "startWithQueue:completionHandler: must be called from main thread."

Reply:
Good catch. This reminds me that we have an `MGLAssertIsMainThread()` macro, but I think we do want to raise an exception explicitly (across all build configurations) instead of potentially only asserting in Debug.